pageserver/src/layered_repository.rs

```diff
@@ -177,7 +177,7 @@ impl Repository for LayeredRepository {
             ancestor_lsn: start_lsn,
         };
         std::fs::create_dir_all(self.conf.timeline_path(&dst, &self.tenantid))?;
```

Comment:
I think we might also need to worry about fsync-ing the parent directories if they're being created

Reply:
Good point. It depends on the OS and mount options, but in principle Linux at least requires is.

That's easy to forget. Perhaps we should have a helper function like 'create_dir_all_and_fsync` and use that everywhere instead of `create_dir_all`.

Reply:
In theory, it's even worse than that: https://www.postgresql.org/message-id/Pine.OSF.4.61.0410311246390.238375%40kosh.hut.fi. That was a long time ago, and we decided back then that that's just crazy, and you should use mount options to prevent that craziness. Modern Linux and filesystem versions don't have that problem AFAIK.